tools/build-variable-schemas.py

```diff
@@ -0,0 +1,135 @@
+    return new_vars
+  
+def create_original_reviewed_variants(variables):
+    # creates original and reviewed variants
+    return \
+        variables + \
+        create_variable_variant(variables=variables , modifier="prefix", affix="original") + \
+        create_variable_variant(variables=variables , modifier="prefix", affix="reviewed")
+
+
+# Create lists with variants
+
+## Dates
+dates = create_original_reviewed_variants(dates_input)
+
+## Names
+names = create_original_reviewed_variants(names_input)
+
+## Numbers
+numbers = create_original_reviewed_variants(numbers_input)
+
+## Stings
+strings_full = create_original_reviewed_variants(strings_input)
+
+strings_without_short_full = create_original_reviewed_variants(strings_without_short_input)
+
+strings = \
+    strings_full + \
+    create_variable_variant(variables=strings_full, modifier="suffix", affix="short") + \
+    strings_without_short_full
+
+## Titles
+titles_full = create_original_reviewed_variants(titles_input)
+
+titles = \
+    titles_full + \
+    create_variable_variant(variables=titles_full, modifier="suffix", affix="short")
+    # -main, and -sub variant creation disabled for 1.0.2, uncomment for 1.1
+    # create_variable_variant(variables=titles_full, modifier="suffix", affix="main") + \
+    # create_variable_variant(variables=titles_full, modifier="suffix", affix="sub") 
+    
+
+
+# Stringify the lists for RNC
+
+rnc_join_character = "\n    | "
+
+dates_stringified = rnc_join_character.join('"{0}"'.format(w) for w in dates)
+names_stringified = rnc_join_character.join('"{0}"'.format(w) for w in names)
+numbers_stringified = rnc_join_character.join('"{0}"'.format(w) for w in numbers)
+strings_stringified = rnc_join_character.join('"{0}"'.format(w) for w in strings)
+titles_stringified = rnc_join_character.join('"{0}"'.format(w) for w in titles)
+    
+# Define the RNC template, to assemble the complete RNC file, to pass to Trang.
+
+rnc_template = '''namespace a = "http://relaxng.org/ns/compatibility/annotations/1.0"
+
+
+## Variables
+div {{
+  
+  ## All variables
+  variables = variables.dates | variables.names | variables.standard
+  
+  ## Standard variables
+  variables.standard = variables.numbers | variables.strings | variables.titles
+  
+  ## Date variables
+  variables.dates = 
+    {dates}
+  
+  ## Name variables
+  variables.names = 
+    {names}
+  
+  ## Number variables
+  variables.numbers =
+    {numbers}
+
+  ## Ttile variables
+  variables.titles = 
+    {titles}
+  
+  ## String variables
+  variables.strings =
+    {strings}
+}}
+'''.format(
+    dates = dates_stringified,
+    names = names_stringified,
+    numbers = numbers_stringified,
+    titles = titles_stringified,
+    strings= strings_stringified,
+    )
+
+# Write to csl-variables.rnc
+
+rnc_filename = '../schemas/styles/csl-variables.rnc'
+rnc_variables_file= open(rnc_filename,'w')
+rnc_variables_file.write(rnc_template)
+rnc_variables_file.close()
```

Comment:
there's newline warning here; could you run the script though, say, black?

Reply:
Better now?